core/sql/arkcmp/CmpStoredProc.cpp

```diff
@@ -273,77 +273,75 @@ NABoolean CmpSPOutputFormat::SetFormat(Lng32 nCols,
   nCols_ = nCols;
   // set up tableDesc_, allocate the storage from allocate routine,
   // so the delete of the tableDesc_ could be done correctly. 
-  // As in sqlcat/desc.h readtabledef_allocate_desc(), allocate routine gets 
+  // As in sqlcat/desc.h TrafAllocateDDLdesc(), allocate routine gets 
   // the storage from HEAP ( i.e. CmpCommon::statementHeap() ). 
   // Likewise, copyString should be the same implementation as in 
   // readRealArk.cpp. It was externalized originally, but being put as
   // static function by someone. Currently the code copyString is cloned
   // in the beginning of the file. ( which will allocate the memory from
   // statementHeap ).
 
-  desc_struct* table_desc = readtabledef_allocate_desc(DESC_TABLE_TYPE);
+  TrafDesc* table_desc = TrafAllocateDDLdesc(DESC_TABLE_TYPE, NULL);
```

Comment:
Why not just make table_desc an object of type TrafTableDesc * and avoid the repetitive casting below?